torchao/dtypes/nf4tensor.py

```diff
@@ -376,7 +376,7 @@ def nf4_pin_memory(aten_op, args, kwargs=None):
     return NF4Tensor(*construct_nf4_args(nf4tensor, updated_attrs))
 
 
-@dataclass
+@dataclass(frozen=True)
```

Comment:
curious how is frozen=True related to PT2?

Reply:
cc @mlazos